tests/Database/DatabaseFunctionalTest.php

```diff
@@ -162,10 +162,26 @@ public function testModifyCollection()
         $createIndexes->execute($this->getPrimaryServer());
 
         $commandResult = $this->database->modifyCollection($this->getCollectionName(), ['index' => ['keyPattern' => ['lastAccess' => 1], 'expireAfterSeconds' => 1000]]);
-
         $this->assertCommandSucceeded($commandResult);
-        $this->assertSame(3, $commandResult['expireAfterSeconds_old']);
-        $this->assertSame(1000, $commandResult['expireAfterSeconds_new']);
+
+        $commandResult = (array) $commandResult;
+
+        if (array_key_exists('raw', $commandResult)) {
+            /* Sharded environment, where we only assert if a shard had a successful update. For
+             * non-primary shards that don't have chunks for the collection, the result contains a
+             * "ns does not exist" error. */
+            foreach ($commandResult['raw'] as $shard) {
+                $shard = (array) $shard;
```

Comment:
Can we make this consistent with ModifyCollectionFunctionalTest by using the following options when calling `modifyCollection()` above?

```
['typeMap' => ['root' => 'array', 'document' => 'array']]
```

Reply:
Done